Blockzilla/BrowserViewController.swift

```diff
@@ -664,16 +751,19 @@ extension BrowserViewController: BrowserToolsetDelegate {
     
     func browserToolsetDidPressBack(_ browserToolset: BrowserToolset) {
         urlBar.dismiss()
+        updateFindInPageVisibility(visible: false)
         webViewController.goBack()
     }
 
     func browserToolsetDidPressForward(_ browserToolset: BrowserToolset) {
         urlBar.dismiss()
+        updateFindInPageVisibility(visible: false)
         webViewController.goForward()
     }
 
     func browserToolsetDidPressReload(_ browserToolset: BrowserToolset) {
         urlBar.dismiss()
+        updateFindInPageVisibility(visible: false)
```

Comment:
Instead of capturing all these separate places in the UI, you could just implement the `func webView(_ webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!)` delegate method for the `WKNavigationDelegate` and do the `urlBar.dismiss()` and `updateFindInPageVisibility(visible: false)` within there.